rtlib/src/SepDisegni.cpp

```diff
@@ -0,0 +1,275 @@
+    namedWindow("Peaks", 0);
+    imshow("Peaks", dist);
+    waitKey(0);
+
+
+
+    // Create the CV_8U version of the distance image
+    // It is needed for findContours()
+    Mat dist_8u;
+    dist.convertTo(dist_8u, CV_8U);
+    // Find total markers
+    vector<vector<Point> > contours;
+    findContours(dist_8u, contours, RETR_EXTERNAL, CHAIN_APPROX_SIMPLE);
+    // Create the marker image for the watershed algorithm
+    Mat markers = Mat::zeros(dist.size(), CV_32S);
+    // Draw the foreground markers
+    for (size_t i = 0; i < contours.size(); i++) {
+        drawContours(markers, contours, static_cast<int>(i), Scalar(static_cast<int>(i) + 1), -1);
+    }
+    // Draw the background marker
+    circle(markers, Point(5, 5), 3, Scalar(255), -1);
+    markers.convertTo(markers, CV_8U);
+    namedWindow("Markers", 0);
+    imshow("Markers", markers * 10000);//FIX ITTT convert to correct C_16F
+    waitKey(0);
+    markers.convertTo(markers, CV_32S);
+
+
+
+    // Perform the watershed algorithm
+    watershed(imgResult, markers);
+    Mat mark;
+    markers.convertTo(mark, CV_8U);
+    bitwise_not(mark, mark);
+    //    imshow("Markers_v2", mark); // uncomment this if you want to see how the mark
+    // image looks like at that point
+    // Generate random colors
+    vector<Vec3b> colors;
+    for (size_t i = 0; i < contours.size(); i++) {
+        int b = theRNG().uniform(0, 256);
+        int g = theRNG().uniform(0, 256);
+        int r = theRNG().uniform(0, 256);
+        colors.push_back(Vec3b((uchar) b, (uchar) g, (uchar) r));
+    }
+    // Create the result image
+    Mat dst = Mat::zeros(markers.size(), CV_8UC3);
+    // Fill labeled objects with random colors
+    for (int i = 0; i < markers.rows; i++) {
+        for (int j = 0; j < markers.cols; j++) {
+            int index = markers.at<int>(i, j);
+            if (index > 0 && index <= static_cast<int>(contours.size())) {
+                dst.at<Vec3b>(i, j) = colors[index - 1];
+            }
+        }
+    }
+    // Visualize the final image
+    namedWindow("Final Result", 0);
+    imshow("Final Result", dst);
+    waitKey(0);
+
+    IntensityImage_ = dst;
+
+
+    return markers;
+}
+
+
+
+
+
+
+vector<Mat> SepDisegni::split_merge(Mat watershededImage) {
+
+
+    /* std::map<int32_t, int32_t> labelCounts;
+    for( int y = 0; y < watershededImage.rows; y++) {
+        for( int x = 0; x < watershededImage.cols; x++) {
+            auto label = watershededImage.at<int32_t>(y,x);
+            if(labelCounts.count(label) > 0) {
+                labelCounts[label] += 1;
+            } else {
+                labelCounts[label] = 1;
+            }
+        }
+    } */
+
+    //Finding Boundary Boxes
+    static const int INT_MINI = std::numeric_limits<int>::min();
+    static const int INT_MAXI = std::numeric_limits<int>::max();
+
+    struct MinMax {
+        cv::Point min{INT_MAXI, INT_MAXI};
+        cv::Point max{INT_MINI, INT_MINI};
+    };
```

Comment:
Put these at the top of the CPP, not in a method, below the includes.

MinMax probably isn't the most descriptive name for future programmers. Maybe "MinMaxTracker" or something?